### DISCUSSION TITLE
No pull requests from other branches to main

### DISCUSSION THREAD

**Discussion by cristianramos9** [uprm-inso4101-2024-2025-s1/semester-project-regiupr#68] [Milestone 2 Kick Offs]
Hi.

While browsing the code of the project, it is easy to note that most of the code is in different branches.
Being split into different branches, it is hard to put together the project as a whole and test it.

I request that all code that was done for Milestone 1 to be sent to 'main' branch as a pull request.

**Comment by Yamil-Serrano** [comment]
Good idea. Does this mean that for the next milestone each team will make their own branch and when milestone 2 is completed everything will be merged? Or will there be another method for each team to manage their part of the work?

**Comment by cristianramos9** [comment]
I believe branches should be created on a need basis.
One use of branches, and the one I think the group should use for the project, is to work on features separate from the main code.
Pull from 'main' to a new branch, work on feature, push to 'main' when feature is complete.
For that, we need to have all code on main.

Here is what I propose we do for the following milestones:

1. First, get all code that is considered "final" pushed to 'main' branch. Once done, everyone will have access to the "complete" code, and can test that it runs on their computers. Here potential bugs can surface.
2. Once 'main' is up to date, each team creates a new branch for the team (team branch).
3. Additionally, if the team have, for example, two features, then two new branches are created for each feature (feature branch).
4. If no multiple features on the team, the previous step can be omitted.
5. From the team branch, or feature branch, each team member creates a branch clone on their computer, works on its tasks, and push completed tasks to the branch.
6. Team leader approves pull requests to the team branch, and then push these changes to 'main'.
7. Other teams pull (or fetch) from main to their branches and makes sure there are no problems.

Branches names must follow a common convention.
For example: 
`TeamName-FeatureName`
Current branches should be named in this convention, except for 'main'.
